### PR TITLE
[YAML] Unexpected characters using YAML compact syntax

### LINKED ISSUE #37788: Unexpected characters using YAML compact syntax
**Symfony version(s) affected**: 5.1.3

**Description**  
I am reading the following YAML file, but I get a parser exception.

**How to reproduce**  

This is the code:

            $config = Yaml::parseFile($ps_filepath, Yaml::PARSE_CONSTANT);

I am reading the following YAML file:

```yaml
macro_scalar : /usr/local/fish
macro_list : [flounder, lobster, haddock]
macro_assoc : { 
  fish : [flounder, cod, haddock],
  shellfish : [scallop, crab, clam],
  other : [chicken, pig, cow]
}

override_value: "global"

```

This is the error:

```
Unexpected characters near "

override_value: "global"" at line 10.
```

If I move line `override_value: "global"` to the begining of the file, the problem dissapears:

```yaml
override_value: "global"
macro_scalar : /usr/local/fish
macro_list : [flounder, lobster, haddock]
macro_assoc : { 
  fish : [flounder, cod, haddock],
  shellfish : [scallop, crab, clam],
  other : [chicken, pig, cow]
}
```

### PR DESCRIPTION
**Pull request by RevZer0**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #37788
| License       | MIT
| Doc PR        | not available

Issue fix

### CODE REVIEW

File: src/Symfony/Component/Yaml/Tests/ParserTest.php [hunk 1]
```diff
@@ -2414,6 +2414,53 @@ public function testParseValueWithNegativeModifiers()
             $this->parser->parse($yaml)
         );
     }
+
+    /**
+     * Test that covers #37788 issue.
```
Comment (fabpot) [inline src/Symfony/Component/Yaml/Tests/ParserTest.php:2419]: Can be removed

File: src/Symfony/Component/Yaml/Parser.php [hunk 2]
```diff
@@ -217,7 +218,7 @@ private function doParse(string $value, int $flags)
                 }
 
                 if (!\is_string($key) && !\is_int($key)) {
-                    throw new ParseException(sprintf('%s keys are not supported. Quote your evaluable mapping keys instead.', is_numeric($key) ? 'Numeric' : 'Non-string'), $this->getRealCurrentLineNb() + 1, $this->currentLine);
+                    throw new ParseException(sprintf('"%s" keys are not supported. Quote your evaluable mapping keys instead.', is_numeric($key) ? 'Numeric' : 'Non-string'), $this->getRealCurrentLineNb() + 1, $this->currentLine);
```
Comment (fabpot) [inline src/Symfony/Component/Yaml/Parser.php:221]: To be reverted (same for other changes suggested by fabbot, they are false positives)

**Review by fabpot — CHANGES_REQUESTED** [review]

**Comment by xabbuh** [comment]
I think we should investigate if this overlaps with #33763.

**Comment by RevZer0** [comment]
> I think we should investigate if this overlaps with #33763.

Haven't seen this pr tbh. As far as I see your update will fix this issue

**Comment by fabpot** [comment]
Does it mean that we can close this one in favor of #33763?

**Comment by RevZer0** [comment]
@fabpot I think yes

**Comment by xabbuh** [comment]
Thank you for working on this @RevZer0. I have added your test cases to #33763.

**Pull request closed without merging**